core/src/main/scala/org/apache/spark/scheduler/DecommissionInfo.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+/**
+ * Provides more detail about a decommissioning event.
+ * @param message Human readable reason for why the decommissioning is happening.
+ * @param isWorkerDecommissioned Whether the worker is being decommissioned too.
+ *                               Used to know if the shuffle data might be lost too.
+ */
+private[spark]
+case class DecommissionInfo(message: String, isWorkerDecommissioned: Boolean)
```

Comment:
so this PR is just a refactor and doesn't actually use the `isWorkerDecommissioned` flag?

Reply:
Yeah. Its actually used in https://github.com/apache/spark/pull/29014 (which builds atop this one). I pulled the usage part to that PR since it was more closely coupled there.